### PR TITLE
Update DeveloperGuide.md

### PR DESCRIPTION
**Pull request by frederickpek**
Update details to match Track2Gather, including,
* target user profile
* value proposition
* user stories
* use cases
* non-functional requirements
* glossary

### CODE REVIEW

**Comment by frederickpek** [comment]
This PR is yet to be finalised.
Perhaps hold on merging until more members add their parts to the the doc.

File: docs/DeveloperGuide.md [hunk 1]
```diff
@@ -255,58 +255,119 @@ _{Explain here how the data archiving feature will be implemented}_
...
-| `* * *`  | user                                       | delete a person                | remove entries that I no longer need                                   |
-| `* * *`  | user                                       | find a person by name          | locate details of persons without having to go through the entire list |
-| `* *`    | user                                       | hide private contact details   | minimize chance of someone else seeing them by accident                |
-| `*`      | user with many persons in the address book | sort persons by name           | locate a person easily                                                 |
+| ------- | ---------------------------------------- | ------------------------------ | ----------------------------------------------------------------------- |
+| `* * *` | new user                                 | see usage instructions         | refer to instructions when I forget how to use the app                  |
+| `* * *` | user                                     | update contacts’ contact details without re-adding that contact| avoid re-entering existing data         |
```
Comment (wei2912) [inline docs/DeveloperGuide.md:275]: There's a missing space before the 2nd last `|` symbol.

File: docs/DeveloperGuide.md [hunk 2]
```diff
@@ -255,58 +255,119 @@ _{Explain here how the data archiving feature will be implemented}_
...
+    
+    Use case ends.
+* 3a. The given index is invalid.
+  * 3a1. Track2Gather shows an error message.
+      
+      Use case resumes at step 2.
+* 3a. The given fields to edit are invalid.
+  * 3a1. Track2Gather shows an error message.
+      
+      Use case resumes at step 2.
```
Comment (wei2912) [inline docs/DeveloperGuide.md:328]: Should this be `4a` instead?

**Review by wei2912 — CHANGES_REQUESTED** [review]
Mostly good but I think there were a few minor typos.

File: docs/DeveloperGuide.md (hunk 1 above)
Comment (frederickpek) [reply docs/DeveloperGuide.md]: Thanks for pointing out! Has been amended.

File: docs/DeveloperGuide.md (hunk 2 above)
Comment (frederickpek) [reply docs/DeveloperGuide.md:328]: It should be 3a too. The `edit` command takes in an index and tags to modify. This extension takes care of invalid inputs to the latter.

Comment (wei2912) [reply docs/DeveloperGuide.md]: As discussed shld be `3b` which I see has been changed alr, closing.

My apologies for the previous typo, my brain was not processing the use case lol

Comment (frederickpek) [reply docs/DeveloperGuide.md]: I have amended it to 3b instead.

**Pull request merged**